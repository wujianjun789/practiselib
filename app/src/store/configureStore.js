import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../common/reducer';
import thunk from 'redux-thunk';


export default function configureStore(initialState) {

    const composeEnhancers =
        process.env.NODE_ENV !== 'production' &&
            typeof window === 'object' &&
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
                // Specify here name, actionsBlacklist, actionsCreators and other options
            }) : compose;

    const enhancer = composeEnhancers(
        applyMiddleware(thunk),
        // other store enhancers if any
    );
    const store = createStore(rootReducer, initialState, enhancer);
    return store;

}
