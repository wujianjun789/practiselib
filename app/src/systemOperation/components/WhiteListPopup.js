import React,{Component} from 'react';
import Panel from '../../components/Panel';
import PanelFooter from '../../components/PanelFooter';
import SearchText from '../../components/SearchText';
import Table from '../../components/Table2';
import Immutable from 'immutable';

import {getWhiteListById, addLcToWhiteListById, delLcFromWhiteListById} from '../../api/domain';

import {getAssetsBaseByModelWithDomain} from '../../api/asset'

export default class WhiteListPopup extends Component {
    constructor(props) {
        super(props);
        this.state = {
            whiteList: [],     /*白名单列表*/
            search: {placeholder: '输入素材名称', value: ''},
            assetsList: [],        /*可添加的设备列表*/
            activeIndex:-1,
        }

        this.columns = [
            {field: "name", title: "名称"},
            {field: "id", title: "编号"},
        ];

        this.onCancel = this.onCancel.bind(this);
        this.onConfirm = this.onConfirm.bind(this);
        this.onChange = this.onChange.bind(this);
        this.onAdd = this.onAdd.bind(this);
        this.searchChange = this.searchChange.bind(this);
        this.itemDelete = this.itemDelete.bind(this);
        this.initWhiteList = this.initWhiteList.bind(this);
        this.updateWhiteList = this.updateWhiteList.bind(this);
        this.initAssetsList = this.initAssetsList.bind(this);
    }

    componentWillMount(){  //需要更新data
        this.mounted = true;
        this.initWhiteList();
        ['lc','screen','xes'].forEach(key=>{
            getAssetsBaseByModelWithDomain(key,this.props.domainId, data =>{
                this.mounted && this.initAssetsList(data)
            });
        })  
    }

    componentWillUnmount(){
        this.mounted = false;
    }

    initWhiteList() {
        const {id} = this.props;
        getWhiteListById(id, data=>{ 
            this.mounted && this.updateWhiteList(data)
            this.props.callFun && this.props.callFun();
        });
    }

    updateWhiteList(data){
        let whiteList = data.map(item=>{
            return {id:item.id,name:item.name}
        })
        this.setState({whiteList:whiteList});
    }

    onAdd() {                  //向whitelist中添加需要的数据,然后更新列表视图
        const {search, assetsList} = this.state
        let curItem = null
        for(var key in assetsList){
            if(assetsList[key].name == search.value){
                curItem = assetsList[key];                
            }
        }
        if(curItem == null){
            return;
        }
        let gatewayId = this.props.id;  //灯集中控制器的id
        let lcId = curItem.id;   
        addLcToWhiteListById(gatewayId, lcId, ()=>{
            this.initWhiteList()
            this.searchChange('');
        })
        
    }


    itemDelete(e) {   //从whiteList中删除资产
        let lcId = e.target.id;
        let gatewayId = this.props.id;
        delLcFromWhiteListById(gatewayId, lcId, this.initWhiteList)
    }

    /*获取可添加到白名单的资产列表*/
    initAssetsList(data) {
        let {assetsList} = this.state;
        data.forEach(item=>assetsList.push(item))
        this.setState({assetsList:assetsList});
    }

    onCancel() {
        this.props.overlayerHide();
    }

    onConfirm() {
        this.props.overlayerHide();
        this.props.onConfirm && this.props.onConfirm();
    }

    onChange(e) {
        let id = e.target.id;
        this.setState({[id]: e.target.value});
    }

    searchChange(value){
        const { search } = this.state;
        let newValue = Object.assign({}, search, {value:value});
        this.setState({search:newValue});
    }

    render() {
        let {className='', id} = this.props;
        let {search, whiteList} = this.state;
        let {assetsList} = this.state;   //可添加的资产数据
        let datalist = [];
        for(var key in assetsList){
            let item = assetsList[key];
            let value = search.value;
            if (!value || item.name.indexOf(value)>-1){
                datalist.push({id:item.id, value:item.name})
            }
        }
        let footer = <div className='modal-footer'><button type="button" className='btn btn-primary' onClick={this.onConfirm}>完成</button></div>
            
        return <div className={className}>
            <Panel title='白名单' footer={footer} closeBtn={true} closeClick={this.onCancel}>
                <div className="row search-group">
                    <SearchText IsTip={true} datalist = {datalist} placeholder={search.placeholder}  value={search.value} onChange={this.searchChange}/>
                    <button className="btn btn-primary" onClick={this.onAdd}>添加</button>
                </div>
                <div className="table-list">
                    <ul className="table-header">
                    {
                        this.columns.map((item, index) =>(<li key={index} className="tables-cell">{item.title}</li>))
                    }
                        <li className="tables-cell"></li>
                    </ul>
                    <div className="table-body">
                        <ul>
                        {
                            whiteList.map((item, index) => (
                                <li key={item.id} className="body-row clearfix">
                                {
                                    this.columns.map((subItem, subIndex) => (
                                        <div key={subIndex} className="tables-cell cell-right" title={item[subItem.field]}>{item[subItem.field]}</div>
                                    ))
                                }
                                    <div className="tables-cell">
                                        <span id={item.id} className="glyphicon glyphicon-trash" onClick={this.itemDelete}></span>
                                    </div>
                                </li>
                            ))
                        }
                        </ul>
                    </div>
                </div>
            </Panel>
        </div>
    }
}