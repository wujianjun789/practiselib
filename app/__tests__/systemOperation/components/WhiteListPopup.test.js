import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl-redux';
import configureStore from '../../../src/store/configureStore';

import WhiteListPopup from '../../../src/systemOperation/components/WhiteListPopup';

const store = configureStore();
 describe('<WhiteListPopup />', () => {
    let className="white-popup";
    let data = [
            {id: '00158D0000CABAD5',name:'xxx1'},
            {id: '00158D0000CABAD5',name:'xxx2'},
        ]
    it('render normal', () => {
        const root = shallow(<Provider store={store}>
                <IntlProvider>
                    <WhiteListPopup className={className}/>
                </IntlProvider>
            </Provider>);
        const cmp = root.find('WhiteListPopup');
        root.setState({whiteList:data});
        const {columns} = cmp.instance();
        expect(cmp.find(`.${className}`).length).toBe(1);

        let tHead = cmp.find('.table-header');
        let headCell = tHead.find('.tables-cell');
        expect(headCell.length).toBe(3);
        expect(headCell.at(0).text()).toBe('名称');
        expect(headCell.at(1).text()).toBe('编号');

        let tBody = cmp.find('.table-body');
        let items = tBody.find('.body-row');
        expect(items.length).toBe(data.length);
        
        let item0 = items.at(0);
        let cells = item0.find('.tables-cell');
        expect(cells.length).toBe(3);
        expect(cells.at(0).text()).toBe(data[0][columns[0].field]);
        expect(cells.at(1).text()).toBe(data[0][columns[1].field]);
        expect(cells.at(2).find('span').hasClass('glyphicon glyphicon-trash')).toBeTruthy();

        expect(cmp.find('Panel').props().title).toBe('白名单');
        
        let searchText = cmp.find('SearchText');
        expect(searchText.prop('placeholder')).toBe('输入素材名称');
        expect(searchText.prop('value')).toBe('');
        
        expect(cmp.find('.search-group .btn-primary').text()).toBe('添加');
    });

    it('Behavior interaction', () => {
        let fn = jest.fn();
        let overlayerHide = jest.fn();
        const cmp = mount(<Provider store={store}>
            <IntlProvider>
                <WhiteListPopup overlayerHide={overlayerHide} />
            </IntlProvider>
        </Provider>);

        let searchText = cmp.find('SearchText');
        let searchInput = searchText.find('input');
        searchInput.simulate('change', {target: {value: 'change'}});
        expect(searchText.prop('value')).toBe('change');

    });

    it('snapshot', () => {
        const cmp = renderer.create(<Provider store={store}>
                <IntlProvider>
                    <WhiteListPopup className={className} />
                </IntlProvider>
            </Provider>);
        const tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    });
 })