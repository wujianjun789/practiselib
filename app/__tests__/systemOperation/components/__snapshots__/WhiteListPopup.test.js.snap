exports[`<WhiteListPopup /> snapshot 1`] = `
<div
  className="white-popup">
  <div
    className="modal-content ">
    <div
      className="modal-header">
      <button
        aria-label="Close"
        className="close"
        data-dismiss="modal"
        onClick={[Function]}
        type="button">
        <span
          aria-hidden="true">
          ×
        </span>
      </button>
      <h4
        className="modal-title">
        白名单
      </h4>
    </div>
    <div
      className="modal-body">
      <div
        className="row search-group">
        <div
          className="searchText "
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}>
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="输入素材名称"
            type="search"
            value="" />
          <ul
            className="" />
          <span
            className="glyphicon glyphicon-search"
            onClick={[Function]} />
        </div>
        <button
          className="btn btn-primary"
          onClick={[Function]}>
          添加
        </button>
      </div>
      <div
        className="table-list">
        <ul
          className="table-header">
          <li
            className="tables-cell">
            名称
          </li>
          <li
            className="tables-cell">
            编号
          </li>
          <li
            className="tables-cell" />
        </ul>
        <div
          className="table-body">
          <ul>
            <li
              className="body-row clearfix">
              <div
                className="tables-cell">
                灯集中控制器
              </div>
              <div
                className="tables-cell">
                00158D0000CABAD5
              </div>
              <div
                className="tables-cell">
                <span
                  className="glyphicon glyphicon-trash"
                  id={1}
                  onClick={[Function]} />
              </div>
            </li>
            <li
              className="body-row clearfix">
              <div
                className="tables-cell">
                灯集中控制器
              </div>
              <div
                className="tables-cell">
                00158D0000CABAD5
              </div>
              <div
                className="tables-cell">
                <span
                  className="glyphicon glyphicon-trash"
                  id={2}
                  onClick={[Function]} />
              </div>
            </li>
            <li
              className="body-row clearfix">
              <div
                className="tables-cell">
                灯集中控制器
              </div>
              <div
                className="tables-cell">
                00158D0000CABAD5
              </div>
              <div
                className="tables-cell">
                <span
                  className="glyphicon glyphicon-trash"
                  id={3}
                  onClick={[Function]} />
              </div>
            </li>
            <li
              className="body-row clearfix">
              <div
                className="tables-cell">
                灯集中控制器
              </div>
              <div
                className="tables-cell">
                00158D0000CABAD5
              </div>
              <div
                className="tables-cell">
                <span
                  className="glyphicon glyphicon-trash"
                  id={4}
                  onClick={[Function]} />
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="modal-footer">
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button">
        取消
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button">
        确认
      </button>
    </div>
  </div>
</div>
`;
